pkg/cloud/cloud.go

```diff
@@ -315,12 +340,19 @@ func (c *cloud) DeleteFileSystem(ctx context.Context, fileSystemId string) error
 	}
 
 	var skipFinalBackup *bool
+	var options []*string
 	for _, tag := range fs.Tags {
-		if *tag.Key == SkipFinalBackupTagKey {
-			if tag.Value != nil {
-				skipFinalBackup, _ = util.StringToBoolPointer(*tag.Value)
+		if *tag.Key == OptionsOnDeletionTagKey && tag.Value != nil {
+			// Only add valid deletion options to prevent errors if the user altered the tag value
+			for _, option := range strings.Split(*tag.Value, " ") {
+				if util.Contains(fsx.DeleteFileSystemOpenZFSOption_Values(), option) {
+					options = append(options, aws.String(option))
```

Comment:
Just to point out, if a user manually changes the tags and accidentally misspells one of the options, the other options will still occur.

Not sure how I feel about this. We hold a standard that if a value is invalid, we opt for default behavior. What's your opinion on this?

This is also something we can point out in a doc.

Reply:
I'm fine with that, lets make sure we note it somewhere